lib/main_containers.pm

```diff
@@ -141,6 +141,7 @@ sub load_container_tests {
     }
 
     foreach (split(',\s*', $runtime)) {
+        my $run_args = OpenQA::Test::RunArgs->new();
```

Comment:
Does this really need to happen for each new runtime or should this be before the `foreach` loop?

Reply:
i would say yes. As i remember the design(i am not sure what has changed since then) each instance has some properties. if the two instances share the same `RunArgs` might cause some conflicts. it was depended what are those properties and how are used in general. Having said that, you can have one single `RunArgs` but then it might need changes in other places too.

Reply:
Maybe Felix is right as we still do `$run_args->{runtime} = $runtime` in `load_image_test`, `load_3rd_party_image_test` and in `load_container_engine_test`.

Reply:
AFAIK the `$run_args` are used for publiccloud specific arguments, so those should not change per container runtime.

Reply:
They are also used to pass the runtime to container tests. I will try to move this line out of the `foreach` loop.

Reply:
I would also move this out of the loop. We are setting the runtime argument accordingly in each function.